### PR TITLE
Terms of service

### PR DESCRIPTION
**Pull request by carycheng77**
Thanks Yang!

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @carycheng77 has signed the CLA. Thanks for the pull request!

File: Box.V2/Managers/BoxTermsOfServiceManager.cs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+            return response.ResponseObject;
+        }
+
+        /// <summary>
+        /// Retrieves information about all terms of service
+        /// </summary>
+        /// <param name="tosType">The type of terms of service to be retrieved - managed or external.</param>
+        /// <returns>A a collection of terms of service objects is returned if the id if the user has access.</returns>
+        public async Task<BoxTermsOfServiceCollection<BoxTermsOfService>> GetTermsOfServicesAsync(string tosType = null)
```
Comment (coolboy) [inline Box.V2/Managers/BoxTermsOfServiceManager.cs]: what's the limit on this api?
do we support paging?
what if 10000+ tos returned?

File: Box.V2/Models/BoxTermsOfService.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        /// The status of the terms of service object
+        /// </summary>
+        [JsonProperty(PropertyName = FieldStatus)]
+        public String Status { get; set; }
+
+        /// <summary>
+        /// The enterprise the terms of service object is associated with
+        /// </summary>
+        [JsonProperty(PropertyName = FieldEnterprise)]
+        public BoxEnterprise Enterprise { get; set; }
```
Comment (coolboy) [inline Box.V2/Models/BoxTermsOfService.cs]: is a full enterprise object?

File: Box.V2/Models/BoxTermsOfServiceUserStatuses.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        /// The Terms of Service object
+        /// </summary>
+        [JsonProperty(PropertyName = FieldTos)]
+        public BoxTermsOfService TermsOfService { get; set; }
+
+        /// <summary>
+        /// The Box user this Terms of Service is associated with
+        /// </summary>
+        [JsonProperty(PropertyName = FieldUser)]
+        public BoxUser User { get; set; }
```
Comment (coolboy) [inline Box.V2/Models/BoxTermsOfServiceUserStatuses.cs]: is this the complete user object?

File: Box.V2/Models/Request/BoxTermsOfServiceUserStatusesRequest.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        /// The Terms Of Service Object
+        /// </summary>
+        [JsonProperty(PropertyName = "tos")]
+        public BoxTermsOfService TermsOfService { get; set; }
+
+        /// <summary>
+        /// The Box User. Default is current Box User.
+        /// </summary>
+        [JsonProperty(PropertyName = "user")]
+        public BoxUser User { get; set; }
```
Comment (coolboy) [inline Box.V2/Models/Request/BoxTermsOfServiceUserStatusesRequest.cs]: Do we need the complete user object here?

File: Box.V2/Models/Request/BoxTermsOfServiceUserStatusesRequest.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    /// <summary>
+    /// A request class for making folder requests
+    /// </summary>
+    public class BoxTermsOfServiceUserStatusesRequest : BoxItemRequest
+    {
+        /// <summary>
+        /// The Terms Of Service Object
+        /// </summary>
+        [JsonProperty(PropertyName = "tos")]
+        public BoxTermsOfService TermsOfService { get; set; }
```
Comment (coolboy) [inline Box.V2/Models/Request/BoxTermsOfServiceUserStatusesRequest.cs]: Do we need the complete object here?

File: Box.V2/Config/BoxConfig.cs [hunk 6]
```diff
@@ -149,6 +149,8 @@ public static IBoxConfig CreateFromJsonString(string jsonString)
         public virtual Uri AuthCodeBaseUri { get { return new Uri(BoxAccountApiHostUri, Constants.AuthCodeString); } }
         public virtual Uri AuthCodeUri { get { return new Uri(AuthCodeBaseUri, string.Format("?response_type=code&client_id={0}&redirect_uri={1}", ClientId, RedirectUri)); } }
         public virtual Uri FoldersEndpointUri { get { return new Uri(BoxApiUri, Constants.FoldersString); } }
+        public virtual Uri TermsOfServicesEndpointUri { get { return new Uri(BoxApiUri, Constants.TermsOfServicesString);  } }
+        public virtual Uri TermsOfServiceUserStatusesUri { get { return new Uri(BoxApiUri, Constants.TermsOfServiceUserStatusesString); } }
```
Comment (coolboy) [inline Box.V2/Config/BoxConfig.cs]: EndpointUri?

**Pull request merged**